### DISCUSSION TITLE
[Question] How to implement the micropip

### DISCUSSION THREAD

**Discussion by lawchin** [flet-dev/flet#1323] [Q&A]
### Question

You can use Micropip API directly in your Flet app:

https://flet.dev/docs/guides/python/publishing-static-website/

I dont know what I did wrong. and I dont know if I need or when to use pyodide & micropip.

### Code sample

```python
if sys.platform == "emscripten":
    import micropip
    await micropip.install("regex")
```
```


### Error message

```bash
await micropip.install("regex")
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
SyntaxError: 'await' outside function
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
Faced this same issue with a project some day. Forgot the package I was trying to import though.

Please @FeodorFitsner, make it clear.

**Comment by FeodorFitsner** [comment]
OK, as [`micropip.install`](https://micropip.pyodide.org/en/v0.2.2/project/api.html#micropip.install) is async method by its nature you need to call it from async method. Therefore, Flet app must be async: https://flet.dev/docs/guides/python/async-apps

For example:

```python
import flet as ft

async def main(page: ft.Page):

    if sys.platform == "emscripten":
        import micropip
        await micropip.install("regex")

    await page.add_async(ft.Text("Hello, async world!"))

ft.app(main)
```

**Reply by lawchin** [reply]
ok thank you very much.. I hope it works this time

**Comment by FeodorFitsner** [comment]
Alternatively, add `regex` to `requirements.txt` on a separate line.

**Reply by lawchin** [reply]
ok will try all of that .. thanks again
